core/src/miner/miner.rs

```diff
@@ -491,7 +491,7 @@ impl Miner {
             if let Some(seal_bytes) = seal.seal_fields() {
                 open_block.seal(self.engine.borrow(), seal_bytes).expect("Sealing always success");
             } else {
-                panic!("Seal should not be none")
+                return Err(BlockError::InvalidSeal.into())
```

Comment:
`InvalidSeal` usually used for block validating functions. How about returning `Some(None)`?

Reply:
You mean changing the function signature to return `Result<Option<(ClosedBlock, Option<H256>)>, Error>`

Reply:
Yes

Reply:
I changed it.